src/jit/codegenxarch.cpp

```diff
@@ -6426,6 +6428,14 @@ void CodeGen::genIntToIntCast(GenTreeCast* cast)
                 break;
 #ifdef _TARGET_64BIT_
             case GenIntCastDesc::ZERO_EXTEND_INT:
+                // We can skip emitting this zero extending move if the previous instruction zero extended implicitly
+                if ((srcReg == dstReg) && (emit->emitCurIGinsCnt > 0) && compiler->opts.OptimizationEnabled())
```

Comment:
I'm not fond of leaking `emitLastIns` out of the emitter. Maybe this should call a function like emitIsLastInsCall(), say `emitIsLastInsZeroExtendingWrite(srcReg)`. Then this code could be:
```
canSkip = (srcReg == dstReg) && compiler->opts.OptimizationEnabled() && emit->emitIsLastInsZeroExtendingWrite(srcReg);
```

Reply:
Sure, let me update.

Reply:
`emitAreUpper32bitZero` might be better, it doesn't matter how they ended up being zero, just that they are.

Reply:
Agree -- I'll go with that.